File: client/lib/card.dart
import 'package:flutter/material.dart';

enum Suit { hearts, diamonds, clubs, spades }

enum Rank { seven, eight, nine, ten, jack, queen, king, ace }

class Card {
  final Suit suit;
  final Rank rank;

  Card({required this.suit, required this.rank});

  @override
  String toString() {
    return '${rankToString(rank)} of ${suitToString(suit)}';
  }
}

Map<String, dynamic> toJson(Card card) {
  return {
    'suit': suitToString(card.suit),
    'rank': rankToString(card.rank),
  };
}

Suit suitFromString(String suit) {
  return Suit.values.firstWhere((s) => s.toString().split('.').last == suit);
}

Rank rankFromString(String rank) {
  return Rank.values.firstWhere((r) => r.toString().split('.').last == rank);
}

String rankToString(Rank rank) {
  switch (rank) {
    case Rank.seven:
      return 'seven';
    case Rank.eight:
      return 'eight';
    case Rank.nine:
      return 'nine';
    case Rank.ten:
      return 'ten';
    case Rank.jack:
      return 'jack';
    case Rank.queen:
      return 'queen';
    case Rank.king:
      return 'king';
    case Rank.ace:
      return 'ace';
    default:
      return rank.toString().split('.').last;
  }
}

String suitToString(Suit suit) {
  return suit.toString().split('.').last;
}

class CardWidget extends StatelessWidget {
  final Card card;

  const CardWidget({super.key, required this.card});

  @override
  Widget build(BuildContext context) {
    String imagePath = getCardImagePath(card);
    double screenWidth = MediaQuery.of(context).size.width;
    double cardWidth;
    double cardHeight;
    if (screenWidth < 750) {
      cardWidth = screenWidth / 8.75;
      cardHeight = cardWidth * 1.65;
    } else {
      cardWidth = screenWidth / 16;
      cardHeight = cardWidth * 1.5;
    }
    return SizedBox(
      width: cardWidth,
      height: cardHeight,
      child: Image.asset(
        imagePath,
        fit: BoxFit.cover,
      ),
    );
  }
}

String getCardImagePath(Card card) {
  return 'assets/${rankToString(card.rank)}_${suitToString(card.suit)}.png';
}

File: serveur/MCTS/main/Card.py
from enum import Enum

class Rank(Enum):
    SEVEN = 7
    EIGHT = 8
    NINE = 9
    TEN = 10
    JACK = 11
    QUEEN = 12
    KING = 13
    ACE = 14

class Suit(Enum):
    CLUBS = 'Clubs'
    DIAMONDS = 'Diamonds'
    SPADES = 'Spades'
    HEARTS = 'Hearts'

class Card:
    def __init__(self, suit, rank):
        self.suit = suit
        self.rank = rank
        self.shorthand = self.set_shorthand()

    def set_shorthand(self):
        rank_dict = {
            Rank.SEVEN: "7",
            Rank.EIGHT: "8",
            Rank.NINE: "9",
            Rank.TEN: "10",
            Rank.JACK: "J",
            Rank.QUEEN: "Q",
            Rank.KING: "K",
            Rank.ACE: "A"
        }
        suit_dict = {
            Suit.SPADES: "\u2660",
            Suit.HEARTS: "\u2661",
            Suit.DIAMONDS: "\u2662",
            Suit.CLUBS: "\u2663"
        }
        return rank_dict[self.rank] + suit_dict[self.suit]
    
    def get_suit(self):
        return self.suit

    def get_rank(self):
        return self.rank

    def print_card(self):
        return f"{self.rank} of {self.suit}"

    def print_card_short(self):
        return self.shorthand
    
    def __lt__(self, other):
        if self.suit == other.get_suit():
            return self.rank.value < other.get_rank().value
        return self.suit.value < other.get_suit().value

    def __eq__(self, other):
        return self.suit == other.get_suit() and self.rank == other.get_rank()

    def copy(self):
        return Card(self.suit, self.rank)